### PR TITLE
Support for rendering values from older to newer on generic widgets

### PR DESCRIPTION
**Pull request by icrc-jofrancisco**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide).
- [ ] My work includes tests or is validated by existing tests.

## Summary
As a developer, I want it to be possible to use a configuration to invert the order of the values. By inverting the data, it will be possible to render the values from the oldest to the most recent.

## Screenshots

Before:
image

After (with dateAxisReverse config set to true):
image

## Related Issue
None
## Other

None

### CODE REVIEW

File: packages/esm-generic-patient-widgets-app/src/config-schema.ts [hunk 1]
```diff
@@ -11,6 +11,11 @@ export const configSchema = {
     _description: 'Displayed in messages about this data',
     _default: 'results',
   },
+  dateAxisReverse: {
```
Comment (ibacher) [inline packages/esm-generic-patient-widgets-app/src/config-schema.ts:14]: Maybe call it: `graphOldestFirst`

File: packages/esm-generic-patient-widgets-app/src/obs-graph/obs-graph.component.tsx [hunk 2]
```diff
@@ -33,17 +33,20 @@ const ObsGraph: React.FC<ObsGraphProps> = ({ patientUuid }) => {
...
-        .filter((obs) => obs.conceptUuid === selectedConcept.uuid && obs.dataType === 'Number')
-        .map((obs) => ({
-          group: selectedConcept.label,
-          key: formatDate(new Date(obs.issued), { year: false, time: false }),
-          value: obs.valueQuantity.value,
-        })),
-    [obss, selectedConcept.uuid, selectedConcept.label],
-  );
+  const chartData = useMemo(() => {
+    let chartRecords = obss
```
Comment (ibacher) [inline packages/esm-generic-patient-widgets-app/src/obs-graph/obs-graph.component.tsx:37]:
```suggestion
    const chartRecords = obss
```

File: packages/esm-generic-patient-widgets-app/src/obs-graph/obs-graph.component.tsx [hunk 3]
```diff
@@ -33,17 +33,20 @@ const ObsGraph: React.FC<ObsGraphProps> = ({ patientUuid }) => {
...
+      .filter((obs) => obs.conceptUuid === selectedConcept.uuid && obs.dataType === 'Number')
+      .map((obs) => ({
+        group: selectedConcept.label,
+        key: formatDate(new Date(obs.issued), { year: false, time: false }),
+        value: obs.valueQuantity.value,
+      }));
+
+    if (config?.graphOldestFirst) {
```
Comment (ibacher) [inline packages/esm-generic-patient-widgets-app/src/obs-graph/obs-graph.component.tsx:45]:
```suggestion
    if (config.graphOldestFirst) {
```

**Review by ibacher — APPROVED** [review]

**Comment by gracepotma** [comment]
@ibacher I see you approved this PR already - were you waiting on anything in particular to Squash & Merge?

Quick question as well - what was the use case for this feature? Reason I ask is, we were really concerned about keeping these charts consistent, because of the potential for users to just glance at the trendline and assume a Left-to-Right timeline, without double-checking the X-axis. So we had tried to keep L2R chart layout consistent. 
Is the main use case to support R2L languages?

**Comment by icrc-jofrancisco** [comment]
The idea here is to be able to see the data from the oldest to the most recent values. If this PR is accepted, this option is only available if the configuration is set up for it. In theory, if you don't change the configuration (as it is now), the user keeps with the same view. however, if the "graphOldestFirst" property is set to "true", the graph will be sorted chronologically (old to new). 
This implementation was requested for the ICRC. 
I hope I have answered the question.

Thanks,

**Comment by icrc-jofrancisco** [comment]
Thanks for the great clarification @ibacher !
Quick answer to your question: No.
However, the code that was made is just inverting the order of the values, I don't think it's a problem, The performance should be the same.
Thanks,

**Pull request merged**